tests/Domain/FundTest.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace MatchBot\Tests\Domain;
+
+use MatchBot\Domain\CampaignFunding;
+use MatchBot\Domain\ChampionFund;
+use MatchBot\Domain\Money;
+use MatchBot\Domain\Salesforce18Id;
+use MatchBot\Tests\TestCase;
+
+class FundTest extends TestCase
+{
+    /**
+     * @dataProvider amountsToSummarise
+     */
+    public function testGetAmounts(Money $totalAmount, Money $amountAvailable, Money $expectedUsedAmount): void
+    {
+        $fund = new ChampionFund(
+            'GBP',
+            'testGetAmounts fund',
+            Salesforce18Id::of('sfFundId4567890abc'),
+        );
+        $fund->addCampaignFunding(new CampaignFunding(
+            fund: $fund,
+            amount: $totalAmount->toNumericString(),
+            amountAvailable: $amountAvailable->toNumericString(),
+            allocationOrder: 100,
+        ));
+
+        $expected = [
+            'totalAmount' => $totalAmount,
+            'usedAmount' => $expectedUsedAmount,
+        ];
+
+        self::assertEquals($expected, $fund->getAmounts());
+    }
+
+    public function testToAmountUsedUpdateModel(): void
+    {
+        $fund = new ChampionFund('GBP', 'Testfund', Salesforce18Id::of('sfFundId4567890abc'));
+        $fund->addCampaignFunding(new CampaignFunding(
+            fund: $fund,
+            amount: '123.45',
+            amountAvailable: '100.00',
+            allocationOrder: 100,
+        ));
+
+        $expected = [
+            'currencyCode' => 'GBP',
+            'fundId' => null, // Not actually persisting it.
+            'fundType' => 'championFund',
+            'salesforceFundId' => 'sfFundId4567890abc',
+            'totalAmount' => '123.45',
+            'usedAmount' => '23.45',
+        ];
+
+        self::assertEquals($expected, $fund->toAmountUsedUpdateModel());
+    }
+
+    /**
+     * Test data provider.
+     *
+     * @return list{array{
+     *     amountAvailable: Money,
+     *     expectedUsedAmount: Money,
+     *     totalAmount: Money,
+     * }}
+     */
+    private function amountsToSummarise(): array
+    {
+        /**
+         * @var list{array{
```

Comment:
Psalm seems adamant that there is a mismatch between the implicitly detected types and the `@return` unless this is specified here, gave up trying to work out why. It doesn't complain of any issues at the point of assignment now so maybe a bug?